server/polar/integrations/open_collective/service.py

```diff
@@ -0,0 +1,103 @@
+import contextlib
+import dataclasses
+from collections.abc import AsyncGenerator
+
+import httpx
+
+from polar.config import settings
+
+
+@dataclasses.dataclass
+class OpenCollectiveCollective:
+    slug: str
+    host_slug: str
+    isActive: bool
+    isApproved: bool
+    isArchived: bool
+    isFrozen: bool
+
+    @property
+    def is_eligible(self) -> bool:
+        return (
+            self.isActive
+            and self.isApproved
+            and not self.isArchived
+            and not self.isFrozen
+            and self.host_slug == "opensource"
+        )
+
+
+class OpenCollectiveServiceError(Exception):
+    def __init__(self, message: str):
+        self.message = message
+        super().__init__(message)
+
+
+class OpenCollectiveAPIError(OpenCollectiveServiceError):
+    pass
+
+
+class CollectiveNotFoundError(OpenCollectiveServiceError):
+    def __init__(self, slug: str):
+        super().__init__(f"No collective found with slug {slug}.")
+
+
+class OpenCollectiveService:
+    def __init__(self, personal_token: str | None = None) -> None:
+        self.personal_token = personal_token
+
+    def create_dashboard_link(self, slug: str) -> str:
+        return f"https://opencollective.com/{slug}"
+
+    async def get_collective(self, slug: str) -> OpenCollectiveCollective:
+        async with self._get_graphql_client() as client:
```

Comment:
This performs a GraphQL query to retrieve a collective from its slug. We have different exceptions to handle different error cases.